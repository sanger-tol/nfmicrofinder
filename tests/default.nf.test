nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("Test pipeline with default parameters") {

        when {
            params {
                outdir = "$outputDir"
                input = "https://ftp.ensembl.org/pub/release-114/fasta/taeniopygia_guttata/dna/Taeniopygia_guttata.bTaeGut1_v1.p.dna.primary_assembly.1.fa.gz"
                pep_file = "https://tolit.cog.sanger.ac.uk/test-data/birds_protein/MicroFinder_prot_set.v0.1.fa.gz"
                scaffold_length_cutoff = 500000
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nfmicrofinder_software_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
