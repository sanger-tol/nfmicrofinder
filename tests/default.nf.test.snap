{
    "Test pipeline with default parameters": {
        "content": [
            4,
            {
                "MICROFINDER_FILTER": {
                    "coreutils": 9.1
                },
                "MINIPROT_ALIGN": {
                    "miniprot": "0.11-r234"
                },
                "MINIPROT_INDEX": {
                    "miniprot": "0.11-r234"
                },
                "RENAME_FALLBACK_OUTPUT": {
                    "coreutils": 9.1
                },
                "Workflow": {
                    "sanger-tol/nfmicrofinder": "v1.0.0"
                }
            },
            [
                "microfinder",
                "microfinder/bTaeGut1.MicroFinder.order.tsv",
                "microfinder/final.fa",
                "miniprot",
                "miniprot/MicroFinder_prot_set.v0.1.fa_aligned.gff",
                "miniprot/Taeniopygia_guttata.bTaeGut1_v1.p.dna.primary_assembly.1.fa.mpi",
                [
                    "execution_trace_2025-09-16_21-19-59.txt:md5,8ad4a81ac400f067841cbfb3f18e6889",
                    "params_2025-09-16_21-20-05.json:md5,9d287b14ba2040c988b7d5ec3209a6c4"
                ],
                "pipeline_info/nfmicrofinder_software_versions.yml"
            ],
            [
                "bTaeGut1.MicroFinder.order.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "final.fa:md5,cef9ba9b0ee7934e24dbc8888f901966",
                "MicroFinder_prot_set.v0.1.fa_aligned.gff:md5,10b706880b50e0df5ae0459a7213e07a",
                "Taeniopygia_guttata.bTaeGut1_v1.p.dna.primary_assembly.1.fa.mpi:md5,b5825c8386cc3338685902e999e7088c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-23T22:07:40.59490712"
    }
}